packages/components/src/dateInput/DateInput.js

```diff
@@ -1,90 +1,124 @@
-  +onDateChange: (value: ?Date) => void,
-  +date: ?Date,
-  +defaultDate: Date,
-  +dateFormat: string,
-  +confirmLabel: string,
-  +cancelLabel: string,
-  +placeholder?: string,
-  +iconName?: IconNameType,
+type Props = {|
+  +onDateChange: (string, string) => void,
+  +date: {
+    +day: string,
+    +month: string,
+    +year: string,
+  },
+  +errors: {
+    +day: string,
+    +year: string,
+  },
 |};
 
-type State = {|
-  isPickerVisible: boolean,
+type Month = {|
+  value: string,
+  label: string,
 |};
 
-export default class DateInput extends React.Component<Props, State> {
-  state = {
-    isPickerVisible: false,
+class DateInput extends React.Component<Props> {
+  getMonthOptions = () => {
+    return [...Array(12).keys()].map<Month>(
+      (currentMonth: number): Month => ({
+        value: String(currentMonth),
+        label: format(setMonth(new Date(), currentMonth), 'MMMM'),
+      }),
+    );
   };
 
-  showDatePicker = () => this.setState({ isPickerVisible: true });
+  handleDayChange = (day: string) => {
+    this.props.onDateChange(day, 'day');
+  };
 
-  hideDatePicker = () => this.setState({ isPickerVisible: false });
+  handleMonthChange = (month: ?string) => {
+    if (month) {
+      this.props.onDateChange(month, 'month');
+    }
+  };
 
-  handleDateChange = (date: ?Date) => {
-    this.props.onDateChange(date);
-    this.hideDatePicker();
+  handleYearChange = (year: string) => {
+    this.props.onDateChange(year, 'year');
   };
 
   render() {
-    const { isPickerVisible } = this.state;
-    const {
-      label,
-      formLabelContainerStyle,
-      formLabelTextStyle,
-      date,
-      defaultDate,
-      dateFormat,
-      placeholder,
-      iconName,
-      confirmLabel,
-      cancelLabel,
-    } = this.props;
-
     return (
-      <View>
-        {label != null && (
-          <FormLabel
-            containerStyle={formLabelContainerStyle}
-            labelStyle={formLabelTextStyle}
-          >
-            {label}
-          </FormLabel>
-        )}
-        <PickerButton
-          value={date ? format(date, dateFormat) : null}
-          placeholder={placeholder}
-          onPress={this.showDatePicker}
-          iconName={iconName}
-        />
-        <DatePicker
-          isVisible={isPickerVisible}
-          labels={{
-            cancel: cancelLabel,
-            confirm: confirmLabel,
-          }}
-          date={date ?? defaultDate}
-          onConfirm={this.handleDateChange}
-          onDismiss={this.hideDatePicker}
-        />
+      <View style={styles.container}>
+        <Text style={styles.text}>Date of birth</Text>
+        <View style={styles.dateFormWrapper}>
+          <View style={styles.dayPickerWrapper}>
+            <TextInput
+              onChangeText={this.handleDayChange}
+              placeholder="DD"
+              autoCorrect={false}
+              value={this.props.date.day}
+              error={this.props.errors.day}
+              confirmLabel="OK"
```

Comment:
I think this prop should not be here, because `TextInput` from UC doesn't have it.

Reply:
It was complaining without it, so I added it. You are asking about `confirmLabel` , no?